sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -627,6 +627,7 @@ quotedIdentifier
 number
     : DECIMAL_VALUE            #decimalLiteral
     | SCIENTIFIC_DECIMAL_VALUE #scientificDecimalLiteral
+    | MINUS INTEGER_VALUE      #negativeIntegerLiteral
     | INTEGER_VALUE            #integerLiteral
```

Comment:
Why not add change this case into: `MINUS? integerLiteral      #integerLiteral`. That also works and this would save quite a code in the `AstBuilder`.